File: Sorting/quickSort.py
def partition(a,low,high):
    i = low-1 
    for j in range(low,high):
        if( a[j] < a[high]):
            i+=1
            a[i],a[j] = a[j],a[i]
    a[high],a[i+1]=a[i+1],a[high]
    return i+1

def quickSort(a,low,high):
    if(low < high):
        pi = partition(a,low,high)
        quickSort(a,low,pi-1)
        quickSort(a,pi+1,high)

a=[5,4,9,12,22,3,7,1000,-90]
quickSort(a,0,len(a)-1)
print(a)

File: Sorting/quickSort.java
public class quickSort {
    public void swap(int a[], int i, int j)
    {
        int t = a[i];
        a[i] = a[j];
        a[j] = t;
    }
    public int partition(int a[], int low, int high)
    {
        int i = low - 1;
        for ( int j=low; j<high ; j++)
        {
            if( a[j] < a[high] )
            {
                i++;
                swap(a, i, j);
            }
        }
        swap(a, i+1, high);
        return i+1;
    }
    public void qSort(int a[], int low, int high)
    {
        if(low < high)
        {
            int pi = partition(a, low, high);
            qSort(a, pi+1 , high);
            qSort(a, low, pi-1);
        }

    }
    public static void main(String args[])
    {
        int a[] = {100,-23,234,5,43,433};
        quickSort qs = new quickSort();
        qs.qSort(a, 0 ,a.length-1);
        for(int i=0;i<a.length;i++)
        {
            System.out.print(a[i]+" ");
        }
    }
}
